src/compiler/resolutionCache.ts

```diff
@@ -1224,12 +1199,9 @@ export function createResolutionCache(resolutionHost: ResolutionCacheHost, rootD
 
     function canWatchTypeRootPath(nodeTypesDirectory: string) {
         // If type roots is specified, watch that path
-        if (resolutionHost.getCompilationSettings().typeRoots) return true;
-
-        // Otherwise can watch directory only if we can watch the parent directory of node_modules/@types
-        const dir = getDirectoryPath(getDirectoryPath(nodeTypesDirectory));
-        const dirPath = resolutionHost.toPath(dir);
-        return dirPath === rootPath || canWatchDirectoryOrFile(dirPath);
+        return !!resolutionHost.getCompilationSettings().typeRoots ||
+            // Otherwise can watch directory only if its at rootDir or we can watch the node_modules directory

```

Comment:
```suggestion
            // Otherwise we'll only watch this path if it falls within `rootDir` or
            // the path is not disqualified by other criteria ("not `C:\Users\Name\Dir`").
```